### ISSUE TITLE
Firebase 5.6.0 dependencies conflict with other Google library dependencies.

### ISSUE THREAD

**Issue by alexmx** [firebase/firebase-ios-sdk#1736]
Describe your environment

  * Xcode version: **9.4.1**
  * Firebase SDK version: **5.6.0**
  * Firebase Component: **Core**
  * Component version: **5.6.0**
  * NearbyMessages version: **1.1.0**

Describe the problem

#### Steps to reproduce:

We can't install latest version of Firebase iOS SDK (5.6.0) using CocoaPods because it's dependencies conflict with NearbyMessages (iOS Nearby SDK) library (also from Google).
We use both Firebase and NearbyMessages (iOS Nearby SDK) libraries in our app.

After we update our Firebase version to 5.6.0, CocoaPods reports that it can't resolve the dependencies.
We get the following error:

```
[!] CocoaPods could not find compatible versions for pod "GoogleUtilities/NSData+zlib":
  In Podfile:
    Firebase (~> 5.6.0) was resolved to 5.6.0, which depends on
      Firebase/Core (= 5.6.0) was resolved to 5.6.0, which depends on
        FirebaseAnalytics (= 5.1.1) was resolved to 5.1.1, which depends on
          GoogleUtilities/NSData+zlib (~> 5.2)

Specs satisfying the `GoogleUtilities/NSData+zlib (~> 5.2)` dependency were found, but they required a higher minimum deployment target.
```

If we change the order in which dependencies are defined we get the following error, so we can see which version GoogleUtilities is used by NearbyMessages.
```
[!] CocoaPods could not find compatible versions for pod "GoogleUtilities":
  In Podfile:
    NearbyMessages (~> 1.1.0) was resolved to 1.1.0, which depends on
      GoogleUtilities (~> 1.3)

Specs satisfying the `GoogleUtilities (~> 1.3)` dependency were found, but they required a higher minimum deployment target.
```

Both Firebase and NearbyMessages are the most recent versions and both use GoogleUtilities library, but completely different versions. Firebase depends on GoogleUtilities `"~> 5.2"` whereas NearbyMessages uses version `"~> 1.3"`.

NearbyMessages seems to be very outdated, the latest version (1.1.0) has been released two years ago. Seems like it's NearbyMessages SDK problem but as it doesn't have a GitHub repo where I can fill an issue, I report it here considering that both frameworks are offered by Google.

**Comment by paulb777** [comment]
Thanks for the report @alexmx.  We'll follow up with the NearbyMessages team and investigate a solution. 

In the meantime a workaround would be to use a version of Firebase less than 5.5.0.

**Comment by paulb777** [comment]
Fixed with NearbyMessages (1.1.1)
